libraries/chain/database.cpp

```diff
@@ -500,6 +500,23 @@ bool database::push_block(const signed_block& new_block, uint32_t skip)
    return result;
 }
 
+void database::_maybe_warn_multiple_production( const database& db, uint32_t height )const
+{
+   auto blocks = _fork_db.fetch_block_by_number( height );
+   if( blocks.size() > 1 )
+   {
+      vector< std::pair< account_name_type, fc::time_point_sec > > witness_time_pairs;
+      for( const auto& b : blocks )
+      {
+         witness_time_pairs.push_back( std::make_pair( b->data.witness, b->data.timestamp ) );
+      }
+
+      ilog( "Encountered a block num collision due to a fork. Walking the current fork to determine the correct block. block_num:${n}", ("n", height) );
```

Comment:
We should combine these messages and make the logging accurate. Something like, "Encountered block num collision at block ${n} produced by witnesses ${w}" This also removes the statement that we are walking the fork, which we are not doing at this point in the code.